graphql_client_codegen/src/lib.rs

```diff
@@ -137,24 +143,34 @@ pub fn generate_module_token_stream(
     Ok(modules)
 }
 
-#[derive(Debug, Error)]
+#[derive(Debug)]
 enum ReadFileError {
-    #[error(
-        "Could not find file with path: {}\
-        Hint: file paths in the GraphQLQuery attribute are relative to the project root (location of the Cargo.toml). Example: query_path = \"src/my_query.graphql\".",
-        path
-    )]
-    FileNotFound {
-        path: String,
-        #[source]
-        io_error: io::Error,
-    },
-    #[error("Error reading file at: {}", path)]
-    ReadError {
-        path: String,
-        #[source]
-        io_error: io::Error,
-    },
+    FileNotFound { path: String, io_error: io::Error },
+    ReadError { path: String, io_error: io::Error },
+}
+
+impl Display for ReadFileError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ReadFileError::FileNotFound { path, .. } => {
+                write!(f, "Could not find file with path: {}\
+                Hint: file paths in the GraphQLQuery attribute are relative to the project root (location of the Cargo.toml). Example: query_path = \"src/my_query.graphql\".", path)
+            }
+            ReadFileError::ReadError { path, .. } => {
+                f.write_str("Error reading file at: ")?;
+                f.write_str(path)
+            }
+        }
+    }
+}
+
+impl std::error::Error for ReadFileError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            ReadFileError::FileNotFound { path: _, io_error } => Some(io_error),
+            ReadFileError::ReadError { path: _, io_error } => Some(io_error),
```

Comment:
This can be collapsed:

```suggestion
            ReadFileError::FileNotFound { io_error, .. }
            | ReadFileError::ReadError { io_error, .. } => Some(io_error),
```